### DISCUSSION TITLE
Connectivity with custom API/endpoint

### DISCUSSION THREAD

**Discussion by matheospower** [open-webui/open-webui#2086] [Ideas]
**Is your feature request related to a problem? Please describe.**
I'm experimenting with various LLM pipelines and would like to use open-webui as the chat interface. To achieve this, I'd like to request a feature that allows connecting the chat interface to my custom pipeline API/endpoint.

**Describe the solution you'd like**
Ideally, I'd want to connect to my custom pipeline through an API by providing a URL (e.g., http://localhost:4217/) and a name for the assistant (e.g., "Custom Assistant"). The generated text would then be displayed in the chat UI.

**Describe alternatives you've considered**
I've attempted to simulate connections with my custom endpoint as an Ollama or OpenAI API, but haven't been successful so far. Guidance on what changes are required or a guide on creating a compatible custom API would be helpful.

**Additional context**
I've searched through GitHub issues and Discord channels but couldn't find an exact match for this request. If someone could provide guidance on how to proceed, I'd be happy to contribute to the solution.

**Comment by jay-jjwu** [comment]
I'm looking for the same feature!

**Reply by maimas** [reply]
Such functionality would be nice - it will allow developers to create their backend services exposed through chat endpoints connected to the web interface.

**Reply by justinh-rahb** [reply]
@maimas [Pipelines Server](https://github.com/open-webui/pipelines) or [Pipe Functions](https://openwebui.com/functions) can fill this need.

**Comment by sunilswain-esspl** [comment]
Any updates on this?

**Comment by chernistry** [comment]
Same here—trying to link OpenWebUI with my custom RAG pipeline using a local OpenAI-compatible v1 endpoint (e.g., http://localhost:8910/), but it’s not getting picked up. Wondering if support for this is planned?
